internal/config/config.go

```diff
@@ -150,13 +177,63 @@ func (c *Config) NewFabricClientForFramework(ctx context.Context, meta tfsdk.Con
 // newFabricClient returns the base fabricv4 client that is then used for either the sdkv2 or framework
 // implementations of the Terraform Provider with exported Methods
 func (c *Config) newFabricClient(ctx context.Context) *fabricv4.APIClient {
+	authClient := c.createAuthClient(ctx)
+
+	// Configure HTTP client with retries and logging
+	httpClient := c.configureHTTPClient(authClient)
+
+	return c.createFabricClient(httpClient)
+}
+
+func (c *Config) createAuthClient(ctx context.Context) *http.Client {
+	if c.StsAuthScope != "" && c.StsSourceToken != "" {
+		// If the StsAuthScope and StsSourceToken are set, use STS based authentication
+		return c.createStsIdentityClient(ctx, c.StsSourceToken)
+	}
+
+	return c.createOAuthClient(ctx)
+}
+
+func (c *Config) createStsIdentityClient(ctx context.Context, stsSourceToken string) *http.Client {
+	httpClient := &http.Client{Timeout: c.requestTimeout()}
+
+	refreshSource := &stsRefreshTokenSource{
+		ctx:            ctx,
+		authScope:      c.StsAuthScope,
+		stsSourceToken: stsSourceToken,
+		client:         httpClient,
+		stsBaseURL:     c.StsBaseURL,
+	}
+
+	tokenSource := xoauth2.ReuseTokenSource(nil, refreshSource)
+
+	return &http.Client{
+		Transport: &xoauth2.Transport{
+			Source: tokenSource,
+		},
+	}
+}
+
+func (c *Config) createOAuthClient(ctx context.Context) *http.Client {
+	if c.Token != "" {
+		tokenSource := xoauth2.StaticTokenSource(&xoauth2.Token{AccessToken: c.Token})
+		oauthTransport := &xoauth2.Transport{
+			Source: tokenSource,
+		}
+		var authClient = &http.Client{
+			Transport: oauthTransport,
+		}
+		return authClient
+	}
 	authConfig := oauth2.Config{
 		ClientID:     c.ClientID,
 		ClientSecret: c.ClientSecret,
 		BaseURL:      c.BaseURL,
 	}
-	authClient := authConfig.New(ctx)
-	//nolint:staticcheck // We should move to subsystem loggers, but that is a much bigger change
+	return authConfig.New(ctx)
+}
+
+func (c *Config) configureHTTPClient(authClient *http.Client) *http.Client {
```

Comment:
We can't address the linter's complaint about `logging.NewTransport` in this PR, so the `nolint` comment that used to precede this line should be brought back.

```suggestion
func (c *Config) configureHTTPClient(authClient *http.Client) *http.Client {
    //nolint:staticcheck // We should move to subsystem loggers, but that is a much bigger change
```